### PR TITLE
Splitting data to train/test set

### PR DESCRIPTION
**Pull request by sibagherian**
Defining a dataset and dataloader pytorch object and splitting data to train/test data set.
feeding data of train/test data by dataloader and being able to compare train data loss against test data loss.
by defining dataloader now implementing mini batch and stochastic gradient descent is possible.

### CODE REVIEW

**Comment by AndreWeiner** [comment]
Dear @sibagherian,
thank you very much for your contribution. I've checked your changes. The path classification notebook is mostly for educational purposes - to understand the "why" behind selected algorithms. I greatly appreciate your efforts to implement a `Dataloader`-derived class and the train-test-split. Unfortunately, the dataset and the classification problem are not suitable for demonstrating the usefulness of splitting. There is simply too little data. Therefore, I'll reject the pull request. However, I keep it in mind for upcoming changes.

A technical comment regarding the pull request: in general, it is recommended practice to
- create a new branch on your fork
- make your changes on that branch
- make a pull request for the working branch (not master or main)

Thanks again for your contribution,
Andre

**Comment by sibagherian** [comment]
Dear @AndreWeiner 
Thanks for your attention and for your comment
being honest I know that it recommended way to create a branch for pull request but I really did not now that even you will check or even you can see my request because of some limitation on my country, by the way for sure you are right and I will try to obey recommended practice!

**Comment by sibagherian** [comment]
Dear @AndreWeiner
Thanks for your attention and for your comment
being honest I know that it recommended way to create a branch for pull request but I really did not know that even you will check or even you can see my request because of some limitation on my country, by the way for sure you are right and I will try to obey recommended practice!
